sqlmesh/core/linter/rules/builtin.py

```diff
@@ -274,4 +275,22 @@ def create_fix(self, model_name: str) -> t.Optional[Fix]:
         )
 
 
+class ValidateModelDefinition(Rule):
+    """
+    Checks whether a model satisfies certain properties, such as (but not limited to):
+
+    - If SQL-based, it contains at least one projection & projection names are unique
+    - Its kind is configured correctly (e.g., the VIEW kind is not supported for Python models)
+    - Other metadata properties are well-formed (e.g., incremental-by-time models require a time column)
+    """
+
+    def check_model(self, model: Model) -> t.Optional[RuleViolation]:
+        try:
+            model.validate_definition()
```

Comment:
let's extract the logic out here, and not have it in validate_definition,

let's change validate definition lint into a different name, like duplicate names and inferrable projections etc

Reply:
Sounds good

Reply:
+1 to this. Ideally we want some granular rules here. One per validation